src/tools/r2rdump/XmlDumper.cs

```diff
@@ -0,0 +1,309 @@
+                {
+                    Serialize(transition, gcNode);
+                }
+
+                if (_raw)
+                {
+                    DumpBytes(method.GcInfo.Offset, (uint)method.GcInfo.Size, methodNode, false);
+                }
+            }
+
+            XmlNode rtfsNode = null;
+            rtfsNode = _xmlDocument.CreateNode("element", "RuntimeFunctions", "");
+            methodNode.AppendChild(rtfsNode);
+
+            foreach (RuntimeFunction runtimeFunction in method.RuntimeFunctions)
+            {
+                DumpRuntimeFunction(runtimeFunction, rtfsNode);
+            }
+        }
+
+        /// <summary>
+        /// Dumps one runtime function. 
+        /// </summary>
+        internal override void DumpRuntimeFunction(RuntimeFunction rtf, XmlNode parentNode)
+        {
+            XmlNode rtfNode = _xmlDocument.CreateNode("element", "RuntimeFunction", "");
+            parentNode.AppendChild(rtfNode);
+            AddXMLNode("MethodRid", rtf.Method.Rid.ToString(), rtfNode);
+            Serialize(rtf, rtfNode);
+
+            if (_disasm)
+            {
+                string disassembly = CoreDisTools.GetCodeBlock(_disassembler, rtf, _r2r.GetOffset(rtf.StartAddress), _r2r.Image);
+                AddXMLNode("Disassembly", disassembly, rtfNode);
+            }
+
+            if (_raw)
+            {
+                DumpBytes(rtf.StartAddress, (uint)rtf.Size, rtfNode);
+            }
+            if (_unwind)
+            {
+                XmlNode unwindNode = null;
+                unwindNode = _xmlDocument.CreateNode("element", "UnwindInfo", "");
+                rtfNode.AppendChild(unwindNode);
+                Serialize(rtf.UnwindInfo, unwindNode);
+
+                if (_raw)
+                {
+                    DumpBytes(rtf.UnwindRVA, (uint)((UnwindInfo)rtf.UnwindInfo).Size, unwindNode);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Prints a formatted string containing a block of bytes from the relative virtual address and size
+        /// </summary>
+        internal override void DumpBytes(int rva, uint size, XmlNode parentNode, bool convertToOffset = true)
+        {
+            int start = rva;
+            if (convertToOffset)
+                start = _r2r.GetOffset(rva);
+            if (start > _r2r.Image.Length || start + size > _r2r.Image.Length)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            if (parentNode != null)
+            {
+                StringBuilder sb = new StringBuilder();
+                sb.Append($"{_r2r.Image[start]:X2}");
+                for (uint i = 1; i < size; i++)
+                {
+                    sb.Append($" {_r2r.Image[start + i]:X2}");
+                }
+                AddXMLNode("Raw", sb.ToString(), parentNode);
+                return;
+            }
+
+            _writer.Write("    ");
```

Comment:
Do you ever expect to hit this code down here? It seems like if a parent node wasn't specified, you wouldn't want to write anything.